metron-platform/metron-hbase/metron-hbase-common/src/test/java/org/apache/metron/hbase/mock/MockHTable.java

```diff
@@ -44,651 +35,757 @@
+            put(put);
+        }
+    }
+
+    @Override
+    public boolean checkAndPut(byte[] row, byte[] family, byte[] qualifier, byte[] value, Put put) throws IOException {
+        return checkAndPut(row, family, qualifier, CompareFilter.CompareOp.EQUAL, value, put);
+    }
+
+    /**
+     * Atomically checks if a row/family/qualifier value matches the expected
+     * value. If it does, it adds the put.  If the passed value is null, the check
+     * is for the lack of column (ie: non-existance)
+     *
+     * @param row       to check
+     * @param family    column family to check
+     * @param qualifier column qualifier to check
+     * @param compareOp comparison operator to use
+     * @param value     the expected value
+     * @param put       data to put if check succeeds
+     * @return true if the new put was executed, false otherwise
+     * @throws IOException e
+     */
+    @Override
+    public boolean checkAndPut(byte[] row, byte[] family, byte[] qualifier, CompareFilter.CompareOp compareOp, byte[] value, Put put) throws IOException {
+        if (check(row, family, qualifier, compareOp, value)) {
+            put(put);
+            return true;
         }
-      }
-    return ret;
-  }
-  public MockHTable(String tableName) {
-    this.tableName = tableName;
-    this.putLog = new ArrayList<>();
-  }
-
-  public MockHTable(String tableName, String... columnFamilies) {
-    this(tableName);
-    for(String cf : columnFamilies) {
-      addColumnFamily(cf);
-    }
-  }
-
-  public int size() {
-    return data.size();
-  }
-
-  public void addColumnFamily(String columnFamily) {
-    this.columnFamilies.add(columnFamily);
-    descriptors = new HColumnDescriptor[columnFamilies.size()];
-    int i = 0;
-    for(String cf : columnFamilies) {
-      descriptors[i++] = new HColumnDescriptor(cf);
-    }
-  }
-
-  public byte[] getTableName() {
-    return Bytes.toBytes(tableName);
-  }
-
-  @Override
-  public TableName getName() {
-    return TableName.valueOf(tableName);
-  }
-
-  @Override
-  public Configuration getConfiguration() {
-    return HBaseConfiguration.create();
-  }
-
-  @Override
-  public HTableDescriptor getTableDescriptor() throws IOException {
-    HTableDescriptor ret = new HTableDescriptor(tableName);
-    for(HColumnDescriptor c : descriptors) {
-      ret.addFamily(c);
-    }
-    return ret;
-  }
-
-  @Override
-  public boolean exists(Get get) throws IOException {
-    if(get.getFamilyMap() == null || get.getFamilyMap().size() == 0) {
-      return data.containsKey(get.getRow());
-    } else {
-      byte[] row = get.getRow();
-      if(!data.containsKey(row)) {
         return false;
-      }
-      for(byte[] family : get.getFamilyMap().keySet()) {
-        if(!data.get(row).containsKey(family)) {
-          return false;
+    }
+
+    @Override
+    public void delete(Delete delete) throws IOException {
+        byte[] row = delete.getRow();
+        if (data.containsKey(row)) {
```

Comment:
Original source versions had extra code for removing family cell maps and the like. Our original implementation also removed this.